### DISCUSSION TITLE
Adding button roles to v3 beta

### DISCUSSION THREAD

**Discussion by JohnGemstone** [orestbida/cookieconsent#403] [Q&A] [answered]
Wondering if v3 (which has a very nice DX with next.js) will ever have config options to assign button roles for the consent modal?
The site I'm working on currently has 'settings' instead of 'reject'. Very much like how it appears on this repo's cover image.
Is this in the roadmap or should I revert to v2?

Thank you!

**Comment by orestbida** [comment, marked as answer]
Hi @JohnGemstone,

In v3 you can set up 1, 2 or 3 button layouts. You can also decide the order and the weight of the buttons. Any layout in v2 can be configured in v3, but not the opposite.

You can check out the 3 button layout here: [playground](https://playground.cookieconsent.orestbida.com/)

**Reply by JohnGemstone** [reply]
Thank you, didn't realise I could just set `acceptNecessaryBtn` to undefined, so easy!

**Reply by orestbida** [reply]
Yes!

I also updated the playground, here's the aforementioned layout in v3:

![eee](https://user-images.githubusercontent.com/34487268/198359437-bcbd057d-f90a-4d3a-95b8-70894a6212cd.PNG)

**Reply by JohnGemstone** [reply]
Best CC library out there! ☕️
